core/eval.go

```diff
@@ -1793,110 +1793,44 @@ func evalPersist(args []string) []byte {
 	return RESP_ONE
 }
 
-// GETEX key [EX seconds | PX milliseconds | EXAT unix-time-seconds |
-// PXAT unix-time-milliseconds | PERSIST]
-// Get the value of key and optionally set its expiration.
-// GETEX is similar to GET, but is a write command with additional options.
-// The GETEX command supports a set of options that modify its behavior:
-// EX seconds -- Set the specified expire time, in seconds.
-// PX milliseconds -- Set the specified expire time, in milliseconds.
-// EXAT timestamp-seconds -- Set the specified Unix time at which the key will expire, in seconds.
-// PXAT timestamp-milliseconds -- Set the specified Unix time at which the key will expire, in milliseconds.
-// PERSIST -- Remove the time to live associated with the key.
-// The RESP value of the key is encoded and then returned
-// evalGET returns RESP_NIL if key is expired or it does not exist
-func evalGETEX(args []string) []byte {
-	if len(args) < 1 {
-		return Encode(errors.New("ERR wrong number of arguments for 'getex' command"), false)
+// evalEXPIREAT sets the expiration time for a specified key.
+// The expiration time is set based on the current time plus the provided seconds.
+// Arguments:
+//
+//	args[0] - The key for which to set the expiration time.
+//	args[1] - The number of seconds from the current time after which the key should expire.
+//
+// Returns:
+//   - Encoded OK response if the operation is successful.
+//   - Encoded error response if the number of arguments is incorrect, if the expiration time is invalid, or if the key does not exist.
+func evalEXPIREAT(args []string) []byte {
+	if len(args) != 2 {
+		return Encode(errors.New("ERR wrong number of arguments for 'EXPIREAT' command"), false)
 	}
 
-	var key string = args[0]
+	key := args[0]
+	seconds, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		return Encode(errors.New("ERR invalid expire time"), false)
+	}
+
+	// Get the current Unix timestamp in milliseconds and add the provided seconds converted to milliseconds
+	expireAtMillis := uint64(time.Now().UnixMilli() + seconds*1000)
```

Comment:
The argument is a unix timestamp, not the time duration. You shouldn't have to add it to the current time.